File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    return findLeastCommonMultiple(arr);
}


const findLeastCommonMultiple = (arr) => {
    let i = 1;
    
    while(1) {
        const count = arr.filter(a => i % a === 0).length;
        if(count === arr.length) break;
        else i++;
    }
    
    return i;
}

/**/

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def is_valid(temp, arr):
    for i in range(1, len(arr)):
        cur = arr[i]
        
        if cur == temp:
            continue
            
        elif temp < cur:
            return False
        
        else:
            if temp % cur == 0:
                continue
            else:
                return False
            
    return True
                

def solution(arr):
    if len(arr) == 1:
        return arr[0]
    
    temp = arr[0]
    i = 1
    while(True):
        if is_valid(temp, arr):
            return temp
        else:
            i += 1
            temp = arr[0] * i
            
            
        